enterprise_access/apps/bffs/context.py

```diff
@@ -43,23 +54,203 @@ def request(self):
+            self._enterprise_customer_uuid = enterprise_customer.get('uuid')
+
+    def _initialize_enterprise_customer_users(self):
+        """
+        Initializes the enterprise customer users for the request user.
+        """
+        try:
+            enterprise_customer_users_data = self.lms_user_api_client.get_enterprise_customers_for_user(
+                self.user.username,
+                traverse_pagination=True
+            )
+        except Exception as e:  # pylint: disable=broad-except
+            self.add_error(
+                user_message='Error retrieving linked enterprise customers',
+                developer_message=str(e)
+            )
+            return
+
+        # Set enterprise features from the response
+        self._enterprise_features = enterprise_customer_users_data.get('enterprise_features', {})
+
+        # Parse the enterprise customer user data
+        enterprise_customer_users = enterprise_customer_users_data.get('results', [])
+        active_enterprise_customer_user = next(
+            (
+                enterprise_customer_user
+                for enterprise_customer_user in enterprise_customer_users
+                if enterprise_customer_user.get('active', False)
+            ),
+            None
+        )
+        active_enterprise_customer = (
+            active_enterprise_customer_user.get('enterprise_customer')
+            if active_enterprise_customer_user else None
+        )
+        enterprise_customer_user_for_requested_customer = next(
+            (
+                enterprise_customer_user
+                for enterprise_customer_user in enterprise_customer_users
+                if self._enterprise_customer_matches_slug_or_uuid(enterprise_customer_user.get('enterprise_customer'))
+            ),
+            None
+        )
+
+        # If no enterprise customer user is found for the requested customer (i.e., request user not explicitly
+        # linked), but the request user is staff, attempt to retrieve enterprise customer metadata from the
+        # `/enterprise-customer/` LMS API endpoint instead.
+        staff_enterprise_customer = None
+        has_enterprise_customer_slug_or_uuid = self.enterprise_customer_slug or self.enterprise_customer_uuid
+        if (
+            not enterprise_customer_user_for_requested_customer and
+            has_enterprise_customer_slug_or_uuid and
+            self.user.is_staff
+        ):
+            try:
+                staff_enterprise_customer = self.lms_api_client.get_enterprise_customer_data(
+                    enterprise_customer_uuid=self.enterprise_customer_uuid,
+                    enterprise_customer_slug=self.enterprise_customer_slug,
+                )
+            except Exception as e:  # pylint: disable=broad-except
+                self.add_error(
+                    user_message='Error retrieving enterprise customer data',
+                    developer_message=str(e)
+                )
```

Comment:
[nit/non-blocking] Another opportunity to chunk logic into a separate function here.
Ex. `get_staff_enterprise_customer`